watcher/lib/kapacitor.go

```diff
@@ -0,0 +1,152 @@
+type KapaClient interface {
+	CreateTask(opt client.CreateTaskOptions) (client.Task, error)
+	Ping() (time.Duration, string, error)
+}
+
+// NewKapaClient creates a Kapacitor client connection
+func NewKapaClient(url, username, password string) (KapaClient, error) {
+	var creds *client.Credentials
+	if username != "" && password != "" {
+		creds = &client.Credentials{
+			Method:   client.UserAuthentication,
+			Username: username,
+			Password: password,
+		}
+	}
+	clt, err := client.New(client.Config{
+		URL:         url,
+		Credentials: creds,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &PaginatingKapaClient{clt, KapacitorFetchRate}, nil
+}
+
+// ensure PaginatingKapaClient is a KapaClient
+var _ KapaClient = &PaginatingKapaClient{}
+
+// PaginatingKapaClient is a Kapacitor client that automatically navigates
+// through Kapacitor's pagination to fetch all results
+type PaginatingKapaClient struct {
+	KapaClient
+	FetchRate int // specifies the number of elements to fetch from Kapacitor at a time
+}
+
+// KapacitorClient communicates to Kapacitor
+type KapacitorClient struct {
+	kapaClient KapaClient
+}
+
+// NewKapacitorClient creates a client that interfaces with Kapacitor tasks
+func NewKapacitorClient() (*KapacitorClient, error) {
+	username := os.Getenv(KapacitorUsernameEnv)
+	password := os.Getenv(KapacitorPasswordEnv)
+
+	kapaClient, err := NewKapaClient(KapacitorAPIAddress, username, password)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &KapacitorClient{
+		kapaClient: kapaClient,
+	}, nil
+}
+
+func (k *KapacitorClient) Health() error {
+	_, _, err := k.kapaClient.Ping()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func (k *KapacitorClient) CreateAlert(name string, script string) error {
+	taskType := client.StreamTask
+	taskTypePipeline := pipeline.StreamEdge
+	if strings.Contains(name, "batch") {
+		taskType = client.BatchTask
+		taskTypePipeline = pipeline.BatchEdge
+	}
+
+	if err := validateTick(script, taskTypePipeline); err != nil {
+		return trace.Wrap(err)
+	}
+
+	dbrps := make([]client.DBRP, 1)
+	dbrps[1] = client.DBRP{
+		Database:        KapacitorDatabase,
+		RetentionPolicy: KapacitorRetentionPolicy,
+	}
+
+	opts := client.CreateTaskOptions{
+		ID:         name,
+		Type:       taskType,
+		DBRPs:      dbrps,
+		TICKscript: script,
+		Status:     client.Enabled,
+	}
+
+	if _, err := k.kapaClient.CreateTask(opts); err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func validateTick(script string, tickType pipeline.EdgeType) error {
+	scope := stateful.NewScope()
+	predefinedVars := map[string]tick.Var{}
+	_, err := pipeline.CreatePipeline(script, tickType, scope, &deadman{}, predefinedVars)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// deadman is an empty implementation of a kapacitor DeadmanService to allow CreatePipeline
+var _ pipeline.DeadmanService = &deadman{}
+
+type deadman struct {
+	interval  time.Duration
```

Comment:
if it's empty - why all these attributes, do you need them?

Reply:
```
// Information relavant to configuring a deadman's swith
type DeadmanService interface {
	Interval() time.Duration
	Threshold() float64
	Id() string
	Message() string
	Global() bool
}
```
I need to implement all of these methods. For sure I can implement something like func Global() { return true }` without have any attribute inside `deadman` struct.